src/guards/doctor.guard.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  HttpException,
+  HttpStatus,
+  Injectable,
+} from '@nestjs/common';
+import { GqlExecutionContext } from '@nestjs/graphql';
+
+import { AuthService } from 'src/modules/auth/auth.service';
+import { Roles } from 'src/types/enums/user-roles.enum';
+
+@Injectable()
+export class DoctorGuard implements CanActivate {
+  constructor(private readonly authService: AuthService) {}
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const ctx = GqlExecutionContext.create(context).getContext();
+    if (!ctx.req.headers.autorization) {
+      return false;
+    }
+    ctx.user = await this.authService.validateToken(
+      ctx.req.headers.autorization,
+    );
+
+    if (ctx.user.role === Roles.DOCTOR || ctx.user.role === Roles.ADMIN) {
+      return true;
+    }
+    throw new HttpException(
+      "You don't have permissions",
+      HttpStatus.UNAUTHORIZED,
+    );
+  }
```

Comment:
Don't you have a similar guards in auth module?

Reply:
I have a guard that checks for an admin role, that one doing it for a doctor, or admin instead (I need more time for research how to use multiple guards on one route so that my lazy implementation)

Reply:
I think you can pass multiple guards to UseGuards decorator as multiple arguments (coma separated).
Also maybe you can also use SetMetadata decorator to set the roles that should be checked and use one guard that will check the metadata values. (as described here https://docs.nestjs.com/guards#setting-roles-per-handler)

Reply:
As I get It right if pass multiple guards they all need to pass. But I need the guard that checking am I a doctor or an admin?

Reply:
Oh, I see, so it's better to use the approach with SetMetadata and one guard to specify what roles to check. (so you won't need to create new one guard for each new case).

Reply:
I had a research and I did not get a reason for using SetMetadata.
If I want to get roles dynamically to set them in setMetadata I need to reinvent guard or use my existing guard recursively.
Mb, I just do not understand how it works. Would you please explain to me how to use it in the right way?